tx/textual/internal/testdata/message.json

```diff
@@ -0,0 +1,139 @@
+[
+    {
+        "proto": {},
+        "screens": [
+            {"text": "Foo object"}
+        ]
+    },
+    {
+        "proto": {
+            "full_name": "nonempty"
+        },
+        "screens": [
+            {"text": "Foo object"},
+            {"text": "Full name: nonempty", "indent": 1}
+        ]
+    },
+    {
+        "proto": {
+            "full_name": "thing one",
+            "nickname": ":thing two"
+        },
+        "screens": [
+            {"text": "Foo object"},
+            {"text": "Full name: thing one", "indent": 1},
+            {"text": "Nickname: :thing two", "indent": 1}
+        ]
+    },
+    {
+        "proto": {
+            "full_name": "special child message",
+            "mtime": {
+                "seconds": 1136214245
+            }
+        },
+        "screens": [
+            {"text": "Foo object"},
+            {"text": "Full name: special child message", "indent": 1},
+            {"text": "Mtime: 2006-01-02T15:04:05Z", "indent": 1}
+        ]
+    },
+    {
+        "proto": {
+            "nickname": "empty child",
+            "left": {}
+        },
+        "screens": [
+            {"text": "Foo object"},
+            {"text": "Nickname: empty child", "indent": 1},
+            {"text": "Left: Foo object", "indent": 1}
+        ]
+    },
+    {
+        "proto": {
+            "nickname": "empty children",
+            "left": {},
+            "right": {},
+            "bar": {}
+        },
+        "screens": [
+            {"text": "Foo object"},
+            {"text": "Nickname: empty children", "indent": 1},
+            {"text": "Left: Foo object", "indent": 1},
+            {"text": "Right: Foo object", "indent": 1},
+            {"text": "Bar: Bar object", "indent": 1}
+        ]
+    },
+    {
+        "proto": {
+            "full_name": "subfield",
+            "left": {},
+            "right": {
+                "nickname": "junior"
+            },
+            "bar": {}
+        },
+        "screens": [
+            {"text": "Foo object"},
+            {"text": "Full name: subfield", "indent": 1},
+            {"text": "Left: Foo object", "indent": 1},
+            {"text": "Right: Foo object", "indent": 1},
+            {"text": "Nickname: junior", "indent": 2},
+            {"text": "Bar: Bar object", "indent": 1}
+        ]
```

Comment:
This is an interesting test case. I'm wondering if the indentation is enough to clearly denote for humans the beginning and end of a struct. From a bijectivity view, yes `indent` is enough, but from a human-readability one, it's almost as if we're expecting the next line to be Bar's fields.

When writing the spec initially, I was pondering on whether to add a `End of Foo object` (like we do in `repeated`) or not. Looking at this test case, I'm considering yes.

Reply:
I'd weigh the utility of the footer screen against the disadvantage of additional clutter. I think it's important to optimize legibility for the common cases, and I doubt that present-but-empty messages will be common. This was mostly meant to explore the corner cases of the renderer than to represent an important test case.